pkg/apis/autoscaling/v1alpha1/pa_lifecycle.go

```diff
@@ -176,18 +176,17 @@ func (pas *PodAutoscalerStatus) IsInactive() bool {
 
 // HasBeenActive returns true if the pod autoscaler has reached its initial scale.
 func (pas *PodAutoscalerStatus) HasBeenActive() bool {
-	if val, ok := pas.Annotations[hasBeenActiveAnnotation]; !ok || val != "true" {
-		return false
-	}
-	return true
+	return pas.GetCondition(PodAutoscalerConditionHasBeenActive).IsTrue()
 }
 
 // MarkHasBeenActive marks the PA's PodAutoscalerConditionInitiallyActive condition true.
 func (pas *PodAutoscalerStatus) MarkHasBeenActive() {
-	if pas.Annotations == nil {
-		pas.Annotations = map[string]string{}
-	}
-	pas.Annotations[hasBeenActiveAnnotation] = "true"
+	podCondSet.Manage(pas).MarkTrue(PodAutoscalerConditionHasBeenActive)
```

Comment:
I presume we have checks in place not to reset? I.e. it's set to `true` only once in the PA lifetime?

Reply:
I updated one of the PA lifecycle tests to make sure marking active and initializing condition set won't affect this status https://github.com/knative/serving/blob/a8d5bbb656c0c199c8c091622518fee4412515a2/pkg/apis/autoscaling/v1alpha1/pa_lifecycle_test.go#L975
I'm not sure where else to place the checks. Do you have any suggestions?

Reply:
Probably in KPA where we evaluate the state. You may just add a test that makes PA ready, then unready then ready again and verify after each step that this condition never changed. 
We have no public methods to set this back, but that doesn't mean we can't directly (also we can probably make PodAutoscalerConditionHasBeenActive private, though it goes against how all other conditions are named).